app/services/automated_workflow_executions/launch_service.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module AutomatedWorkflowExecutions
+  # Service used to Launch an AutomatedWorkflowExecution
+  class LaunchService < BaseService
+    attr_accessor :automated_workflow_execution, :sample, :pe_attachment_pair, :workflow
+
+    def initialize(automated_workflow_execution, sample, pe_attachment_pair, user = nil, params = {})
+      super(user, params)
+      @automated_workflow_execution = automated_workflow_execution
+      @sample = sample
+      @pe_attachment_pair = pe_attachment_pair
+      @workflow = Irida::Pipelines.find_pipeline_by(@automated_workflow_execution.metadata['workflow_name'],
```

Comment:
I think we should return an error if a workflow isn't found? The `workflow_executions_params` would cause an error if it's nil when attempting to access its attributes. And if the workflow isn't found we shouldn't need to call the create service for the workflow execution

Reply:
Fixed in https://github.com/phac-nml/irida-next/pull/574/commits/9fe86f0187a4f2a4afff14bbe0c94111b244ef4d